### PR TITLE
Add relations between existing models

### PR DESCRIPTION
**Pull request by deepakrkris**
Capability to add a relation to a model definition

- [X] - add a loopback model for ModelRelation
- [X] - add a relationship handler to connector
- [X] - add Edge class to graph
- [X] - add a ModelRelation class to data model
- [X] - add method to class Task to add a relation to model definition
- [X] - test cases to add a relation between models

### CODE REVIEW

File: component/datamodel/model.js [hunk 1]
```diff
@@ -55,6 +58,13 @@ class Model extends Node {
       config.ModelDefaultDir, modelDef.name + '.json');
     return filePath;
   }
+  addRelation(relationName, toModelId, data) {
+    const workspace = this._graph;
+    const id = this._name + '.' + relationName;
+    const toModel = workspace.getModel(toModelId);
+    const relation = new ModelRelation(id, this, toModel, data);
```
Comment (superkhau) [inline component/datamodel/model.js]: `return new ModelRelation...` -- It's obvious you're returning a relation.

File: component/tasks.js [hunk 2]
```diff
@@ -31,6 +31,13 @@ class Tasks {
     model.setProperty(propertyName, property);
     cb(null, propertyDef);
   }
+  addModelRelation(relationName, fromModelId, toModelId, data, cb) {
+    const workspace = this;
+    const model = workspace.getModel(fromModelId);
+    const relation = model.addRelation(relationName, toModelId, data);
+    model.setRelation(relationName, relation);
+    writeOperations.writeModel(model, cb);
```
Comment (superkhau) [inline component/tasks.js]: I think we should make your helper util lib name more obvious -- like `fsWriter.writeModel(model, cb)` to make it obvious you're writing to the filesystem.

File: connector/connector.js [hunk 3]
```diff
@@ -24,3 +25,10 @@ connector.createModelProperty = function(modelId, propertyName, data, cb) {
   const workspace = WorkspaceManager.getWorkspace();
   workspace.addModelProperty(modelId, propertyName, data, cb);
 };
+
+connector.createModelRelation =
+  function(relationName, fromModelId, toModelId, data, cb) {
```
Comment (superkhau) [inline connector/connector.js]: indent wrong

```
connector.createModelRelation =
function(relationName, fromModelId, toModelId, data, cb) {
  ...
};
```

File: connector/relation-handler.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+class RelationsHandler {
+  static createRelation(
+    workspace, relationName, fromModelId, toModelId, data, cb) {
+    function create(next) {
+      workspace.addModelRelation(
+        relationName, fromModelId, toModelId, data, function(err) {
```
Comment (superkhau) [inline connector/relation-handler.js]:
```
workspace.addModelRelation(relationName, fromModelId, toModelId, data,
function(err) {
  next(err);
});

File: test/acceptance-tests/use-cases/test-scripts/create-models.js [hunk 5]
```diff
@@ -86,4 +88,49 @@ module.exports = function() {
...
+        id: testsuite.relationName,
+        type: relationType,
+        foreignKey: foreignKey,
+        modelId: testsuite.fromModelName,
+        model: testsuite.toModelName,
+        facetName: 'common',
+      };
+      testsuite.relationTest = {};
+      testsuite.relationTest.facetName = relationDef.facetName;
+      testsuite.relationTest.fromModelName = relationDef.modelId;
```
Comment (superkhau) [inline test/acceptance-tests/use-cases/test-scripts/create-models.js]: Should the above 3 be moved into the callback? Ie. only create these relationTest* variables if the ModelRelation.create succeeds?

File: test/acceptance-tests/use-cases/test-scripts/create-models.js [hunk 6]
```diff
@@ -86,4 +88,49 @@ module.exports = function() {
...
+      ModelRelation.create(relationDef, {}, function(err, data) {
```
Comment (superkhau) [inline test/acceptance-tests/use-cases/test-scripts/create-models.js]: remove unused property `data`

Comment (deepakrkris) [inline test/acceptance-tests/use-cases/test-scripts/create-models.js]: agreed

File: test/acceptance-tests/use-cases/test-scripts/create-models.js (hunk 5 above)
Comment (deepakrkris) [inline test/acceptance-tests/use-cases/test-scripts/create-models.js]: agreed

File: connector/relation-handler.js (hunk 4 above)
Comment (deepakrkris) [inline connector/relation-handler.js]: agreed

File: component/tasks.js (hunk 2 above)
Comment (deepakrkris) [inline component/tasks.js]: agreed

File: component/datamodel/model.js (hunk 1 above)
Comment (deepakrkris) [inline component/datamodel/model.js]: agreed

**Pull request merged**